src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -175,9 +175,9 @@ public partial class Ping
             else
             {
                 cts.Cancel();
-                if (p.ExitCode != 0)
+                if (p.ExitCode == 1 || p.ExitCode == 2)
                 {
-                    // This means no reply was received, although transmission may have been successful.
+                    // Throw timeout for known failure return codes from ping functions.
                     return CreateTimedOutPingReply();
                 }
 
```

Comment:
For the case of segmentation fault, we get the output, the ping is successful, but osx fails after that, so we won't throw at all for that case. For other signal codes, if we don't have output, we will throw later in line 198..